src/jit/optcse.cpp

```diff
@@ -1013,23 +1187,47 @@ void Compiler::optValnumCSE_Availablity()
 
             for (GenTree* tree = stmt->GetTreeList(); tree != nullptr; tree = tree->gtNext)
             {
+                bool isUse = false;
+                bool isDef = false;
+
                 if (IS_CSE_INDEX(tree->gtCSEnum))
                 {
-                    unsigned     CSEnum = GET_CSE_INDEX(tree->gtCSEnum);
-                    unsigned int cseBit = genCSEnum2bit(CSEnum);
-                    CSEdsc*      desc   = optCSEfindDsc(CSEnum);
-                    unsigned     stmw   = block->getBBWeight(this);
-                    bool         isUse  = BitVecOps::IsMember(cseTraits, available_cses, cseBit);
-                    bool         isDef  = !isUse; // If is isn't a CSE use, it is a CSE def
-#ifdef DEBUG
-                    VNFuncApp excSeq;
+                    unsigned CSEnum               = GET_CSE_INDEX(tree->gtCSEnum);
+                    unsigned CseAvailBit          = genCSEnum2bit(CSEnum) * 2;
+                    unsigned cseAvailCrossCallBit = CseAvailBit + 1;
+                    CSEdsc*  desc                 = optCSEfindDsc(CSEnum);
+                    unsigned stmw                 = block->getBBWeight(this);
+
+                    isUse = BitVecOps::IsMember(cseLivenessTraits, available_cses, CseAvailBit);
+                    isDef = !isUse; // If is isn't a CSE use, it is a CSE def
+
+                    // Is this a "use", that we haven't yet marked as live accross a call
+                    // and it is not available when we have calls that kill CSE's (cseAvailCrossCallBit)
+                    // if the above is true then we will mark this the CSE as live across a call
+                    //
+                    bool madeLiveAcrossCall = false;
+                    if (isUse && !desc->csdLiveAcrossCall &&
+                        !BitVecOps::IsMember(cseLivenessTraits, available_cses, cseAvailCrossCallBit))
+                    {
+                        desc->csdLiveAcrossCall = true;
+                        madeLiveAcrossCall      = true;
+                    }
+
+                    // If this a a CSE def, the the call-kill bit should also be a def (i.e not set in the
```

Comment:
typo "the the" should be "then the" I think.
It is confusing to say that "the call-kill bit should also be a def". I think it would be clearer to change this comment to:
```
// If this is a CSE def (i.e. the CSE is not available here, since it is being defined), then the call-kill bit
// should also be zero since it is also not available across a call.
```